### PR TITLE
Kotlin migration: StyledProgressDialog

### PR DESCRIPTION
**Pull request by Arnold2381**
Kotlin migration

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt [hunk 1]
```diff
@@ -16,84 +16,74 @@
...
+            message: CharSequence?,
+            cancelable: Boolean = false,
+            cancelListener: DialogInterface.OnCancelListener? = null
+        ): MaterialDialog {
+            var t = title
+            if ("" == t) {
+                t = null
+                Timber.d("Invalid title was provided. Using null")
+            }
+            return MaterialDialog.Builder(context!!)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt:65]: The `!!`s here appear to be incorrect

**Comment by Arnold2381** [comment]
It failed. David I will check it.
if it doesn't get resolved I will ping you.

**Comment by Arnold2381** [comment]
Its failing. couldnt get it work.

**Comment by david-allison-1** [comment]
This one is a pain. Use this:

```kotlin
/*
 *  Copyright (c) 2021 David Allison <davidallisongithub@gmail.com>
 *
 *  This program is free software; you can redistribute it and/or modify it under
 *  the terms of the GNU General Public License as published by the Free Software
 *  Foundation; either version 3 of the License, or (at your option) any later
 *  version.
 *
 *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
 *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License along with
 *  this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.ichi2.utils

import android.content.DialogInterface
import com.afollestad.materialdialogs.MaterialDialog

// Extension methods for MaterialDialog.Builder workarounds in Kotlin
// Previously the methods accepted null into a @NonNull parameter, 
// and fixing this made the fluent

fun MaterialDialog.Builder.titleNullable(title: CharSequence?): MaterialDialog.Builder {
    title?.let { this.title(it) }
    return this
}

fun MaterialDialog.Builder.contentNullable(content: CharSequence?): MaterialDialog.Builder {
    content?.let { this.content(it) }
    return this
}

fun MaterialDialog.Builder.cancelListenerNullable(cancelListener: DialogInterface.OnCancelListener?): MaterialDialog.Builder {
    cancelListener?.let { this.cancelListener(it) }
    return this
}
```

**Comment by Arnold2381** [comment]
The tests failed in my local, though I have pushed the changes.

File: AnkiDroid/src/main/java/com/ichi2/utils/MaterialBuilderUtil.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+
+package com.ichi2.utils
+
+import android.content.DialogInterface
+import com.afollestad.materialdialogs.MaterialDialog
+
+// Extension methods for MaterialDialog.Builder workarounds in Kotlin
+// Previously the methods accepted null into a @NonNull parameter,
+// and fixing this made the fluent
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/MaterialBuilderUtil.kt:24]:
```suggestion
// and fixing this broke the fluent interface
```

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                .titleNullable(t!!)
+                .contentNullable(message!!)
+                .progress(true, 0)
+                .cancelable(cancelable)
+                .cancelListenerNullable(cancelListener!!)
+                .show()
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt:74]:
```suggestion
            return MaterialDialog.Builder(context!!)
                .titleNullable(t)
                .contentNullable(message)
                .progress(true, 0)
                .cancelable(cancelable)
                .cancelListenerNullable(cancelListener)
                .show()
```

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    fun setProgressStyle() {
+        // TODO
+    }
+
+    companion object {
+        @JvmStatic
+        @JvmOverloads
+        fun show(
+            context: Context?,
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt:57]:
```suggestion
            context: Context,
```

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
This is also required

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt (hunk 3 above)
Comment (Arnold2381) [reply AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt:74]: Still failing

**Comment by Arnold2381** [comment]
I think this should work!

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt [hunk 5]
```diff
@@ -16,84 +16,62 @@
...
-    }
-
-    public void setMax(int max) {
+    fun setMax() {
         // TODO
     }
 
-
-    public void setProgress(int progress) {
+    fun setProgress() {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt:45]: Variables were removed here, instead you should use `Suppress("unused_parameter")

File: AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt [hunk 6]
```diff
@@ -16,84 +16,62 @@
...
-                .title(title)
-                .content(message)
-                .progress(true, 0)
-                .cancelable(cancelable)
-                .cancelListener(cancelListener)
-                .show();
-    }
-
-
-    private static boolean animationEnabled(Context context) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/themes/StyledProgressDialog.kt]: This method was removed, it should be left for a further refactor

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Looks good! Two small issues

**Comment by Arnold2381** [comment]
@david-allison-1 once this goes. I will make changes in the SimpleMessageDialog because it needs the util methods.

**Comment by Arnold2381** [comment]
@david-allison-1 i think u missed this.

**Review by david-allison-1 — APPROVED** [review]
I did miss it! 😱 

Thanks for the reminder! Looks great

**Review by mikehardy — APPROVED** [review]
:zap:

**Pull request merged**